db/go_level_db.go

```diff
@@ -118,12 +118,12 @@ func (db *GoLevelDB) Stats() map[string]string {
 }
 
 type goLevelDBIterator struct {
-	iterator.Iterator
+	source iterator.Iterator
 }
 
 // Key returns a copy of the current key.
 func (it *goLevelDBIterator) Key() []byte {
-	key := it.Key()
+	key := it.source.Key()
```

Comment:
oy can't believe we missed this first time around :(

Reply:
I know right :/